File: Modules/Angular/Day13/Milit-Panchasara/assignment1/src/app/models/student.ts
import { Address } from "./address";
import { EmergencyDetail } from "./emergencyDetail";
import { ParentDetail } from "./parentDetail";
import { ReferenceDetails } from "./referenceDetails";

export interface Student {
    studentName:{
        firstName: string,
        middleName: string,
        lastName: string
    },
    dob: Date,
    placeOfBirth: string,
    firstLanguage: string,
    address: Address,
    father: ParentDetail,
    mother: ParentDetail,
    emergencyDetail: EmergencyDetail[],
    referenceDetail: ReferenceDetails
}

File: Modules/Angular/Day19/Milit-Panchasara/assignmentAPIs/Models/Student.cs
using System;
using System.Collections.Generic;

#nullable disable

namespace assignmentAPIs.Models
{
    public partial class Student
    {
        public Student()
        {
            EmergencyDetails = new HashSet<EmergencyDetail>();
            Parents = new HashSet<Parent>();
            ReferenceDetails = new ReferenceDetail();
        }

        public long Id { get; set; }
        public string FirstName { get; set; }
        public string MiddleName { get; set; }
        public string LastName { get; set; }
        public DateTime DateOfBirth { get; set; }
        public string PlaceOfBirth { get; set; }
        public string FirstLanguage { get; set; }
        public string StreetAddress { get; set; }
        public string City { get; set; }
        public string State { get; set; }
        public string Country { get; set; }
        public string Pin { get; set; }
        public string ContactNumber { get; set; }

        public virtual ICollection<EmergencyDetail> EmergencyDetails { get; set; }
        public virtual ICollection<Parent> Parents { get; set; }
        public virtual ReferenceDetail ReferenceDetails { get; set; }
    }
}
